homeassistant/helpers/entity.py

```diff
@@ -425,6 +428,20 @@ def _name_translation_key(self) -> str | None:
             f".{self.translation_key}.name"
         )
 
+    def _name_placeholders(self, name: str) -> str:
+        """Process possible placeholders in entity name."""
+        tuples = list(string.Formatter().parse(name))
+        if tuples[0][1] is None:
+            return name
+        if (translation_placeholders := self.translation_placeholders) is not None:
+            if TYPE_CHECKING:
+                assert isinstance(translation_placeholders, Mapping)
```

Comment:
Why is this check needed?
`Entity.translation_placeholders` is typed as `Mapping | None` and we've ruled out the `None` case already.

Reply:
The check is removed